_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+
+```bash
+$ helm upgrade kafka-connector openfaas/kafka-connector \
+    --install \
+    --namespace openfaas \
+    --set topics="faas-request\," \
+    --set broker_host="kafka" \
+    --set print_response_body="true"
+```
+
+Here it is important to note that our Broker resides in service called `kafka` in the same namespace as our Gateway and Kafka Connector service. If you have Broker service in another namespace, again called `kafka`, just append the namespace like so `kafka.<your_namespace>` to the `broker_host` environmental variable so the Connector can discover it. The default kafka port `9092` is appended to the service name by default.
+
+Also if you would like to subscribe the connector and your functions to more than one topic, list them in the `topics` variable by separating them with comma and re-run the command above.
+
+Congratulations now you have the whole stack deployed:
+* Zookeeper
+* Broker Host with Producer
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas | grep -E 'kafka|zookeeper|producer'
+kafka-broker-544fbccd48-p6pgr     1/1     Running            3          71m
+kafka-connector-5d9f447f5-5drv7   1/1     Running            0          69m
+zookeeper-699f568f6f-6b4n2        1/1     Running            0          71m
+```
+
+## Subscribe function
+
+In order to consume topics via the connector we simply need to apply an annotation with a key of `topic` and the chosen value used in the `connector-dep.yml` - in our case it was `faas-request`.
+
+Here it is important to note that if you would like your function to subscribe to multiple topics, you just need to list them as we did in the connector configuration.
+
+Before we start log in into your Docker Hub account.
+
+We will first create custom function the following way:
+
+```
+$ faas-cli new kafka-message --lang=go --prefix=<your_dockerhub_username> && mv kafka-message.yml stack.yml
+```
+
+The function is simple Hello World written in Go you can edit it if you want, but for our example we will keep the default message for simplicity.
+
+Edit the `stack.yml` file by adding `topic` annotation with value which we pointed in the Kafka Connector `faas-request` the file should look like this:
+
+```
+provider:
+  name: faas
+  gateway: http://127.0.0.1:8080
+functions:
+  kafka-message:
+    lang: go
+    handler: ./kafka-message
+    image: <your_dockerhub_username>/kafka-message:latest
+    annotations:
+      topic: faas-request
+```
+
+Build, Push and Deploy the function with single command:
+
+```bash
+$ faas-cli up
+```
+
+Voila, your function is now subscribed to the topic and it can be invoked by producing message on that topic.
+
+Now you might ask:
+
+"Great, but how am I supposed see the output of my function?"
+
+Remember in the configuration we set `print_response_body` to `true` so we can see the response body in the logs of the Kafka Connector.
```

Comment:
We cant, not yet

Reply:
I hope to fix this before the blog is out but I can remove it ?

Reply:
Removed the reference